### ISSUE TITLE
responders won't start if a provider failed to start

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1556]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1552
- Created at 2012-09-27 21:14:32 by [pbrezina](https://pagure.io/user/pbrezina)
- Closed as Invalid
- Assigned to nobody
---

The monitor tries to start up the providers first and then starts up the responders if either all providers are running or a timeout has expired. However, if a provider fails to start for whatever reason, every responder will fail to start as well.

This can be easily reproduced by disabling initialization of the provider, for example calling "return 0" immediately in the main() function in data_provider_be.c.

### Comments

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-10-02 12:01:15](https://pagure.io/SSSD/sssd/issue/1552#227572)**

It has been like this at least since 1.5.0.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-10-02 13:58:29](https://pagure.io/SSSD/sssd/issue/1552#227573)**

This is actually more or less intentional. If any configured provider isn't starting up, we want the end-user/local administrator to immediately see the problem. If we just started up in offline mode, the problem could go unnoticed for days or even months (if the cache was sufficiently primed). This could lead to subtle security issues as group memberships fail to be updated.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-10-02 16:58:53](https://pagure.io/SSSD/sssd/issue/1552#227574)**

I was thinking that in case of sssd misconfiguration and in case the users only had sudo access, not root, it might be nice to allow them to log in and fix issues even in case the back ends wouldn't start. Also `services_startup_timeout()` seems to suggest that was the plan at one point:

     DEBUG(1, ("Providers did not start in time, "
               "forcing services startup!\n"));
    
     ctx->services_started = true;
    
     DEBUG(4, ("Now starting services!\n"));
     /* then start all services */
     for (i = 0; ctx->services[i]; i++) {
         add_new_service(ctx, ctx->services[i], 0);
     }

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-10-02 20:31:06](https://pagure.io/SSSD/sssd/issue/1552#227575)**

Even if it is intentional, there is a bug around it.

1. sssd monitor remains running
2. pid file still exists so service status is OK

I'm not really sure if 2 still happens with the recent pidfile patch. It depends whether the responder is terminated before or after it is registered in the monitor.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2012-10-04 15:50:02](https://pagure.io/SSSD/sssd/issue/1552#227576)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.11 beta
rhbz:  => todo

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2013-01-02 15:23:31](https://pagure.io/SSSD/sssd/issue/1552#227577)**

Fields changed

milestone: SSSD 1.11 beta => SSSD 1.12 beta

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-02-16 12:04:37](https://pagure.io/SSSD/sssd/issue/1552#227578)**

This is really important for the secrets-as-a-service work. It's not clear yet whether we will do this work for /all/ responders or only the secrets one, but this ticket should stay in the 1.14 bucket.

blocking:  => #2913
changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
mark:  => 0
review:  => 0
selected:  => 
sensitive:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-06-20 11:31:45](https://pagure.io/SSSD/sssd/issue/1552#227579)**

In the 1.14 milestone, just investigate if this has an impact on the secrets responder. If not, defer.

milestone: SSSD 1.14 beta => SSSD 1.14.0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-06-27 11:08:16](https://pagure.io/SSSD/sssd/issue/1552#227580)**

Fields changed

priority: major => minor

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-07-01 16:08:40](https://pagure.io/SSSD/sssd/issue/1552#227581)**

1.14.0 should be released no later than Wednesday next week, this ticket should not block the 1.14.0 release.

milestone: SSSD 1.14.0 => SSSD 1.14.1

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-08-17 15:47:11](https://pagure.io/SSSD/sssd/issue/1552#227582)**

Moving to triage to discuss if we still need this.

milestone: SSSD 1.14.1 => NEEDS_TRIAGE

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-08-19 16:52:57](https://pagure.io/SSSD/sssd/issue/1552#227583)**

ticket #2243 tracks a more generic solution.

resolution:  => wontfix
status: new => closed

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-09-10 11:28:55](https://pagure.io/SSSD/sssd/issue/1552#227584)**

Fields changed

rhbz: todo => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-02-24 14:36:05](https://pagure.io/SSSD/sssd/issue/1552#227585)**

**Metadata Update from @pbrezina**:
- Issue marked as blocked by: #2913
- Issue set to the milestone: NEEDS_TRIAGE
